src/App.jsx

```diff
@@ -14,45 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_LENGTH = 'sortByLength';
+const SORT_ALPHABET = 'SortAlphabetically';
+
+function sortBy(goods, sortField, rev) {
+  const prepearedGoods = [...goods];
+
+  if (sortField === 'sortByLength' && rev % 2 !== 0) {
+    prepearedGoods.sort((curent, next) => curent.length - next.length);
+  }
+
+  if (sortField === 'sortByLength' && rev % 2 === 0) {
+    prepearedGoods.sort((curent, next) => next.length - curent.length);
+  }
+
+  if (sortField === 'SortAlphabetically' && rev % 2 !== 0) {
+    prepearedGoods.sort((curent, next) => curent.localeCompare(next));
+  }
+
+  if (sortField === 'SortAlphabetically' && rev % 2 === 0) {
+    prepearedGoods.sort((curent, next) => next.localeCompare(curent));
+  }
+
+  if (sortField === '' && rev % 2 === 0) {
+    prepearedGoods.reverse();
+  }
```

Comment:
Looks too complicated and not obvious. Let's try to simplify it using switch/case statement and remove this weird check (ref % 2 !== 0)

Here are an example:

```
  if (prepearedGoods ) {
    prepearedGoods .sort((good1, good2) => {
      switch (sortField) {
        case 'SortAlphabetically':
          return good1.localeCompare(good2);
        case 'sortByLength':
          return good1.length - good2.length;
        default:
      }

      return 0;
    });
  }

  if (isReverse) { // our boolean variable
    filteredGoods = prepearedGoods .reverse();
  }
```